client/state/themes/actions.js

```diff
@@ -420,6 +420,10 @@ export function installTheme( themeId, siteId ) {
 			themeId
 		} );
 
+		if ( isThemeFromWpcom( themeId ) ) {
+			dispatch( installWpcomParentTheme( themeId, siteId ) );
```

Comment:
This part is misleading as it suggests that we always install parent theme. Maybe instead of `installWpcomParentTheme` we should have function that returns parent id if there is one or null and later use that with standard `themeInstall`. Basically what `installWpcomParentTheme` is doing now except dispatching install inside it. This way we would still have one install action available. This way also the recurrence loop will be more visible.